### ISSUE TITLE
ggforest use different ref-level

### ISSUE THREAD

**Issue by PeterStrom** [kassambara/survminer#404]
Hi. Awesome work! would be nice to be able to specify which factor level should be the reference. You could of course do relevel(x, ref=y) but that also changes the order the levels are presented in the plot.

**Comment by kassambara** [comment]
You're right that `relevel(x, ref=)` isn't a satisfying answer here — it changes the reference but also reorders the levels in the plot, which is exactly what you wanted to avoid. I looked for an order-preserving workaround and there isn't a clean one today:

- `relevel()` always moves the new reference to the first position, so `ggforest()` (which lays out rows in `levels()` order) shows it at the top and reorders the rest.
- The usual order-preserving trick, `contrasts(x) <- contr.treatment(k, base = ...)`, currently breaks `ggforest()`: coxph then names the coefficients `x1`, `x2`, ... instead of `xLevelName`, so `ggforest`'s exact coefficient matcher fails and every level renders as "reference" with no hazard ratio.

So setting a custom reference level while keeping the original display order is an unimplemented feature, not just a usage question. Keeping this open as an enhancement — the fix would be either a `ggforest()` argument to pick the reference, or reordering the rows back to the original level order after fitting. Thanks for the clear report.

**Comment by kassambara** [comment]
Quick status and a verified workaround while this stays open as an enhancement.

**Correct workaround today — `relevel()`.** Setting the reference with `relevel()` gives the correct hazard ratios; the only downside is that the chosen reference moves to the top and the other levels shift down (the reordering you wanted to avoid).

```r
library(survminer); library(survival)
d <- lung; d$ph.ecog <- factor(d$ph.ecog); d$sex <- factor(d$sex)
d$ph.ecog <- relevel(d$ph.ecog, ref = "1")   # reference = level 1
ggforest(coxph(Surv(time, status) ~ ph.ecog + sex, data = d), data = d)
```

![relevel: correct HRs vs reference 1, but rows reordered to 1,0,2,3](https://i.imgur.com/Iqo7iqo.png)

**Do not use `contrasts(x) <- contr.treatment(k, base=)` with `ggforest()`.** It keeps the level order, but `coxph()` then names the coefficients by position (`ph.ecog1`, `ph.ecog2`, …) rather than by level, so `ggforest()`'s coefficient matcher mislabels the rows: the intended reference shows a hazard ratio while other levels are wrongly marked "reference". The hazard ratios are computed correctly but attached to the wrong rows — a misleading plot:

![contr.treatment: wrong "reference" labels (0 and 2 marked reference, level 1 shows an HR)](https://i.imgur.com/RPgBFwo.png)

So the feature request — set a custom reference **and** keep the original display order — isn't cleanly available yet. Keeping this open as an enhancement. The likely direction is a `ggforest()` argument to reorder the displayed rows back to a given level order (used together with `relevel()`), rather than teaching the matcher to parse `contr.treatment` coefficient names. Thanks for the report, @PeterStrom.

**Comment by kassambara** [comment]
Keeping this open, and flagging a real defect I under-stated earlier: when a factor uses non-default contrasts (e.g. `contr.treatment(base = 2)`), `ggforest()` currently mislabels the plot — `coxph` names those coefficients positionally (`ph.ecog1`, `ph.ecog3`, `ph.ecog4`), but `ggforest`'s factor branch matches by `paste0(variable, level)`, so the wrong rows get marked as the reference and the HRs land on the wrong levels, with no error. That's a misleading-output bug, not just a limitation.

It's also more fixable than I implied: the multi-coefficient branch already maps coefficients to terms via `model$assign` (the #689 fix), so the factor branch can use the same mapping instead of name-matching. Tracking that here. The `relevel()` route above stays the correct workaround in the meantime (it reorders the displayed rows, which is the original ask).

**Comment by kassambara** [comment]
Fixed in dev (0.5.2.999, #813). `ggforest()` now reads the level-to-coefficient mapping from the model's own contrast matrix, so setting a custom reference the order-preserving way works:

```r
library(survminer); library(survival)
d <- lung
d$ph.ecog <- factor(d$ph.ecog)                       # levels 0, 1, 2, 3
contrasts(d$ph.ecog) <- contr.treatment(4, base = 2) # reference = level 1, order kept
fit <- coxph(Surv(time, status) ~ sex + ph.ecog, data = d)
ggforest(fit, data = d)
```

![ggforest with contr.treatment(base = 2): reference on level 1, order preserved](https://i.imgur.com/PTP6uSL.png)

The reference sits on level 1, the other levels stay in their original order (0, 2, 3), and every hazard ratio lands on the right row — level 0 = 0.66, level 2 = 1.70, level 3 = 5.11 — matching `summary(fit)$conf.int`. Previously this marked two levels as "reference" and dropped the last level's coefficient.

So `contrasts(x) <- contr.treatment(k, base = j)` is now the way to pick a reference level without reordering the plot, which was the original request. `relevel()` still works too if you don't mind the reordering. Thanks again for the report, @PeterStrom.
